lib/song.rb

```diff
@@ -0,0 +1,70 @@
+class Song
+  attr_accessor :name
+  attr_reader :artist
+
+  @@all = []
+
+  def initialize(name, artist = nil, genre = nil)
+    @name = name
+    self.artist = artist if artist
+    self.genre = genre if genre
+  end
+
+  def self.all
+    @@all
+  end
+
+  def save
+    @@all << self
+  end
+
+  def self.create(name)
+    song = new(name)
+    song.save
+    song
+  end
+
+  def self.destroy_all
+    @@all.clear
+  end
+
+  def artist=(artist)
+    @artist = artist
+    artist.add_song(self)
+  end
+
+  def genre
+    @genre
+  end
```

Comment:
Is there a way you can change this into a one liner? What helper method could you use?